ios/Vosk.swift

```diff
@@ -69,25 +87,57 @@ class Vosk: RCTEventEmitter {
         if (currentModel != nil) {
             currentModel = nil; // deinit model
         }
-        currentModel = VoskModel(name: name)
-        resolve(name)
+        
+        // Load the model in a try catch block
+        do {
+            try currentModel = VoskModel(name: name)
+            resolve(true);
+        } catch {
+            reject(nil, nil, nil);
+        }
+    }
+
+
+    /// Set grammar to use
+    @objc(setGrammar:withResolver:withRejecter:)
+    func setGrammar(words: [String]?, resolve:RCTPromiseResolveBlock, reject:RCTPromiseRejectBlock) -> Void {
```

Comment:
No cleaning/ restart needed ? Even after multiple calls/ recognition ? No leak ?

Reply:
Grammar is stored into an internal class variable, for sure it won't work unless you call stop() and start() again. How did you do on Android ? I may have implemented a wrong pattern ^^

Reply:
I clean the recognizer, call the native setGrammar method, and restart, wdy think ?

```kotlin
@ReactMethod
  fun setGrammar(grammar: ReadableArray? = null, promise: Promise) {
    if (recognizer == null || speechService == null) {
      promise.reject(IOException("Recognizer is not started yet"))
    } else {
      try {
        speechService!!.stop()
        speechService!!.shutdown()

        if (grammar != null)
          recognizer!!.setGrammar(makeGrammar(grammar))
        else
          recognizer!!.setGrammar("[]")

        speechService = SpeechService(recognizer, sampleRate)
        if (speechService!!.startListening(this))
          return promise.resolve("Recognizer successfully started")

      } catch (e: IOException) {
        promise.reject(e)
      }
    }
  }
```

Reply:
@kingdcreations : ok I did the same implementation on my side, but it made me think about the timeout argument that id not preserved, isn't it ? I mean, I call start with timeout: 5, then setGrammar(), and my timeout disappear no ?

Reply:
You are right, what should i do ?

Reply:
In a first time, removing setGrammar seem fine no ? Devs can just call stop() and start() again with grammar in parameters, and they keep control of the recognition state on their UI. Isn't it better?

Reply:
Yup, lets do that